lib/logstash/agent.rb

```diff
@@ -127,36 +98,45 @@ def execute
       configure_logging(log_file)
     end
 
-    pipeline.configure("filter-workers", filter_workers)
-
-    # Stop now if we are only asking for a config test.
-    if config_test?
-      report "Configuration OK"
-      return
+    if @config_string || @config_path then
+      @config_string = format_config @config_path, @config_string
+      @pipeline = setup_pipeline
+    else
+      fail(help + "\n" + I18n.t("logstash.agent.missing-configuration"))
     end
 
-    @logger.unsubscribe(stdout_logs) if show_startup_errors
-
-    # TODO(sissel): Get pipeline completion status.
-    pipeline.run
     return 0
-  rescue LogStash::ConfigurationError => e
-    @logger.unsubscribe(stdout_logs) if show_startup_errors
-    report I18n.t("logstash.agent.error", :error => e)
-    if !config_test?
-      report I18n.t("logstash.agent.configtest-flag-information")
-    end
-    return 1
+
   rescue => e
-    @logger.unsubscribe(stdout_logs) if show_startup_errors
     report I18n.t("oops", :error => e)
     report e.backtrace if @logger.debug? || $DEBUGLIST.include?("stacktrace")
     return 1
   ensure
+    @logger.unsubscribe(stdout_logs) if show_startup_errors
     @log_fd.close if @log_fd
     Stud::untrap("INT", trap_id) unless trap_id.nil?
   end # def execute
 
+  def format_config config_path, config_string
```

Comment:
Would add parentheses to this line.
def format_config(config_path, config_string)